viewer/packages/3d-overlays/src/Overlay3DIcon.ts

```diff
@@ -0,0 +1,175 @@
+/*!
+ * Copyright 2022 Cognite AS
+ */
+
+import { EventTrigger } from '@reveal/utilities';
+import * as THREE from 'three';
+import { PerspectiveCamera, Ray, Sphere, Vector3 } from 'three';
+import { clamp } from 'three/src/math/MathUtils';
```

Comment:
Not sure, clamp has some concrete meaning, so I'd say that it's okay to use it. If something changes, it should be noticable easily as it will change the size of icons for raycasting on hovering overlay. Also, it was used in `Image360Icon` for quite a while, this part of the code was not changed at all and adapted from 360 images icons.